django_tasks/backends/database/management/commands/db_worker.py

```diff
@@ -33,16 +34,22 @@ def __init__(
         self.running_task = False
 
     def shutdown(self, signum: int, frame: Optional[FrameType]) -> None:
+        if not self.running:
+            logger.warning(
+                "Received %s - shutting down immediately.", signal.strsignal(signum)
+            )
+            sys.exit(1)
+
         logger.warning(
-            "Received %s - shutting down gracefully.", signal.strsignal(signum)
+            "Received %s - shutting down gracefully... (press Ctrl+C again to force)",
+            signal.strsignal(signum),
         )
-
         self.running = False
 
         if not self.running_task:
             # If we're not currently running a task, exit immediately.
             # This is useful if we're currently in a `sleep`.
-            exit(0)
+            sys.exit(0)
```

Comment:
`exit` is meant for the console; `sys.exit` is meant for code.

Source: https://stackoverflow.com/a/6501134/51801

Reply:
TIL!